admin/client/src/legacy/GridField.js

```diff
@@ -137,18 +137,37 @@ $.entwine('ss', function($) {
   $('.grid-field .action.action_import:button').entwine({
     onclick: function(e) {
       e.preventDefault();
+      this.openmodal();
+    },
+    onmatch: function() {
+			this._super();
+			// Trigger auto-open
+			if (this.data('state') === 'open') {
+			  this.openmodal();
+      }
+		},
+		onunmatch: function() {
+			this._super();
+		},
 
-      var backdrop = $('.modal-backdrop');
+    openmodal: function() {
+      // Remove existing modal
+      let modal = $(this.data('target'));
```

Comment:
I think it's fine, just checking if replacing the modal was intentional, that pjax reloading throws me off sometimes... :/